templates/app_client/lib/app_client.rb.erb

```diff
@@ -64,7 +65,9 @@ module <%= gem_name_pascal %>
     end
 
     def resource_namespace
-      '<%= "#{api_versions.first.resource_namespace}/" if api_versions.first.resource_namespace %>'
+<% if api_versions.first.resource_namespace -%>
+     options[:resource_namespace] || '<%= "#{api_versions.first.resource_namespace}/" %>'
```

Comment:
I see `api_version` wont quite work here but what if you changed it to be

```
def api_version
  if options[:api_version] && options[:resource_namespace]
    OpenStruct.new(name: options[:api_version], options[:resource_namespace])
  else
    api_versions.first
  end
end
```

Then we could get rid of all the other `if`s and just use `api_version.name` and `api_version.resource_namespace`